### PR TITLE
Mz/fix apply lookup table bivariate check

### PR DESCRIPTION
**Pull request by mayeul-zama**
Fixes https://github.com/zama-ai/tfhe-rs-internal/issues/364

### CODE REVIEW

**Review by IceTDrinker — CHANGES_REQUESTED** [review]
Request change for now to avoid merging this name update 😉

**Comment by mayeul-zama** [comment]
We could have a single function `is_functional_bivariate_pbs_possible` which takes an `Option<&BivariatelookupTable>`
This way, if you have a `BivariatelookupTable` at hand, it's logical to give it

**Comment by IceTDrinker** [comment]
> We could have a single function `is_functional_bivariate_pbs_possible` which takes an `Option<&BivariatelookupTable>` This way, if you have a `BivariatelookupTable` at hand, it's logical to give it

I think I like that idea

File: tfhe/src/shortint/server_key/bivariate_pbs.rs [hunk 1]
```diff
@@ -294,13 +290,11 @@ impl ServerKey {
         &self,
         ct1: CiphertextNoiseDegree,
         ct2: CiphertextNoiseDegree,
+        lut: Option<&BivariateLookupTable<Vec<u64>>>,
```
[the remark this answers is not in the document]
Comment (IceTDrinker) [reply tfhe/src/shortint/server_key/bivariate_pbs.rs:293]: We have an Owned put type alias I believe ?

[the remark this answers is not in the document]
Comment (mayeul-zama) [reply tfhe/src/shortint/server_key/bivariate_pbs.rs]: Replaced

File: tfhe/src/shortint/server_key/bivariate_pbs.rs [hunk 2]
```diff
@@ -47,14 +30,11 @@ fn ciphertexts_can_be_packed_without_exceeding_space_or_noise(
 ) -> Result<(), CheckError> {
     let final_degree = (lhs.degree * factor) + rhs.degree;
 
-    let max_degree =
-        MaxDegree::from_msg_carry_modulus(server_key.message_modulus, server_key.carry_modulus);
```
Comment (IceTDrinker) [inline tfhe/src/shortint/server_key/bivariate_pbs.rs:51]: can you check if we have other places where we still do that ?

**Review by IceTDrinker — COMMENTED** [review]
One small question otherwise looks good

**Comment by tmontaigu** [comment]
Looks good to me

File: tfhe/src/shortint/server_key/bivariate_pbs.rs (hunk 2 above)
Comment (mayeul-zama) [reply tfhe/src/shortint/server_key/bivariate_pbs.rs:51]: Use ct MaxDegree as sk MaxDegree may be smaller (to keep carry propagation margin)

Comment (mayeul-zama) [reply tfhe/src/shortint/server_key/bivariate_pbs.rs:51]: Or keep previous code which was correct

Comment (IceTDrinker) [reply tfhe/src/shortint/server_key/bivariate_pbs.rs:51]: indeed with a comment

**Review by IceTDrinker — APPROVED** [review]
Looks good, thanks !

**Pull request merged**